sqsconsumer.go

```diff
@@ -47,30 +48,30 @@ func (m *DefaultSQSQueueConsumer) StartConsuming(ctx context.Context) error {
 			return nil
 		default:
 		}
-		var result, e = m.Queue.ReceiveMessage(&sqs.ReceiveMessageInput{
+		var result, e = m.Queue.ReceiveMessage(ctx, &sqs.ReceiveMessageInput{
 			QueueUrl: aws.String(m.QueueURL),
-			AttributeNames: aws.StringSlice([]string{
+			MessageSystemAttributeNames: []types.MessageSystemAttributeName{
 				"SentTimestamp",
-			}),
-			MessageAttributeNames: aws.StringSlice([]string{
+			},
+			MessageAttributeNames: []string{
 				"All",
-			}),
-			WaitTimeSeconds: aws.Int64(int64(math.Ceil((15 * time.Second).Seconds()))),
-		})
+			},
+			WaitTimeSeconds: int32(math.Ceil((15 * time.Second).Seconds())),
+		}, nil)
 		if e != nil {
-			if !(request.IsErrorRetryable(e) || request.IsErrorThrottle(e)) {
+			if !(m.retrierConfig.IsErrorRetryable(e)) {
```

Comment:
`IsErrorRetryable` now contains logic to handle throttle exceptions as well ([link](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/aws/retry#hdr-Standard))